src/fmm.jl

```diff
@@ -0,0 +1,377 @@
+using ExaFMMt
+using BEAST
+using LinearAlgebra
+using LinearMaps
+using SparseArrays
+using StaticArrays
+
+"""
+    assemble_fmm(
+        spoints::Matrix{F},
+        tpoints::Matrix{F};
+        options=LaplaceFMMOptions() 
+    ) where F <: Real
+
+Function calls setup routine of ExaFMM library. Without specific options, Laplace FMM is used.
+
+# Arguments
+- `spoints::Matrix{F}`: Array of source points in Float64 or Float32.
+- `tpoints::Matrix{F}`: Array of target points in Float64 or Float32.
+- `options::ExaFMMt.FMMOptions`: Specifies FMM type, default is Laplace.
+"""
+function assemble_fmm(
+    spoints::Matrix{F},
+    tpoints::Matrix{F};
+    options=LaplaceFMMOptions() 
+) where F <: Real
+    
+    A = setup(spoints, tpoints, options)
+    return A
+
+end
+
+"""
+    getfullrankblocks(
+        operator::BEAST.AbstractOperator,
+        test_functions::BEAST.Space,
+        trial_functions::BEAST.Space;
+        threading=:single,
+        nmin=10,
+        quadstratcbk=BEAST.DoubleNumQStrat(1,1),
+        quadstratfbk=BEAST.defaultquadstrat(operator, test_functions, trial_functions)
+    )
+
+Function assembles the fullrankblocks that contain the closeinteractions of basis and test functions. 
+These interactions are faulty when computed by the FMM. 
+The faulty values of the FMM are subtracted of the solution using direct evaluations with the same quadrature strategy as used do sample the input points of the FMM.
```

Comment:
Could you limit the line length 92 characters?

I will integrate the JuliaFormatter so that this is enforced in future automatically.